test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1

```diff
@@ -1,184 +1,109 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-Describe "Basic Send-MailMessage tests" -Tags CI {
+
+Describe "Send-MailMessage" -Tags CI, RequireSudoOnUnix {
     BeforeAll {
-        function test-smtpserver
-        {
-            $rv = $false
+        Register-PackageSource -Name nuget.org -Location https://www.nuget.org/api/v2 -ProviderName NuGet -ErrorAction SilentlyContinue
 
-            try
-            {
-                $tc = New-Object -TypeName System.Net.Sockets.TcpClient -ArgumentList "localhost", 25
-                $rv = $tc.Connected
-                $tc.Close()
-            }
-            catch
-            {
-                $rv = false
-            }
+        $nugetPackage = "netDumbster"
+        Find-Package $nugetPackage -ProviderName NuGet | Install-Package -Scope CurrentUser -Force
```

Comment:
It can be slow. @TravisEz13 Can we cache this? If no these tests should have Feature tag.

Reply:
TL;DR:
No caching until at least March:

https://blogs.msdn.microsoft.com/devops/2019/02/05/adding-caching-to-azure-pipelines/

Reply:
@ThreeFive-O Please make the test Feature.

Reply:
Thank you for the comments.

@iSazonov Since the Linux and macOS tests have the `RequireSudoOnUnix` tag, the tests will always be executed as stated in the [test documentation](https://github.com/PowerShell/PowerShell/blob/ee3a8c14c8e16bec4cf89bf3d2ea20ab499acfa9/docs/testing-guidelines/WritingPesterTests.md):

> RequireSudoOnUnix tag is mutually exclusive to all other tags like CI, Feature etc. (which are now ignored when RequireSudoOnUnix is present) and is treated as CI.

The corresponding code in [ci.psm1](https://github.com/PowerShell/PowerShell/blob/ee3a8c14c8e16bec4cf89bf3d2ea20ab499acfa9/tools/ci.psm1#L685) also shows that the tests will always be run on Linux and macOS despite the CI or Feature tag.

On the windows build agent the tests completed in 11m 51s which is similar to all other PR build times. Since the dependency only is required for this specific test and not for other tests, the build agents wouldn't really benefit for caching those, especially since the build agents start of fresh with the next CI run.

In my opinion the tests should run as part of CI context and not as Feature, since the additional time to pull the dependency is negligible and the Send-MailMessage cmdlet tests would always be verified.
Please let me know if running in CI context is fine.